### ISSUE TITLE
Migrate testDeleteWithXXX from BaseConnectorTest to AbstractTestEngineOnlyQueries

### ISSUE THREAD

**Issue by Praveen2112** [trinodb/trino#13210]
`BaseConnectorTest#testDeleteWithXXX` verifies delete operation for various patterns of `DELETE` query - which makes it more suitable to be a part of `AbstractTestEngineOnlyQueries`

**Comment by kokosing** [comment]
@Praveen2112 Can you please elaborate? I am not sure if it is the way to go to migrate these tests.

**Comment by Praveen2112** [comment]
`BaseConnectorTest#testDeleteWithXXX` generally deals with sub-queries (scalar, correlated,) which would be independent of the connector implementation - provided it supports `DELETE` operation

**Comment by Praveen2112** [comment]
`BaseConnectorTest#testDeleteWithXXX` most of the tests validate whether the Trino could execute a specific plan type instead of a `applyDelete` implementation.

**Comment by hashhar** [comment]
Note that since complex expression pushdowns have been added some of them do involve connector as well:

- `testDeleteWithLike`
- `testDeleteWithVarcharPredicate`
- `testDeleteWithComplexPredicate`

**Comment by Praveen2112** [comment]
Yeah !! But some of them like `testDeleteWithSubquery` should be agnostic to the connector impl.

**Comment by Praveen2112** [comment]
It was more like for changes in the Trino planner and optimizer (wrt to sub-query planning) we do write the test code in one of the connector which shouldn't be the case.
